allennlp_models/generation/models/simple_seq2seq.py

```diff
@@ -234,42 +239,46 @@ def forward(
 
     @overrides
     def make_output_human_readable(
-        self, output_dict: Dict[str, torch.Tensor]
+            self, output_dict: Dict[str, torch.Tensor]
     ) -> Dict[str, torch.Tensor]:
         """
         Finalize predictions.
-
         This method overrides `Model.make_output_human_readable`, which gets called after `Model.forward`, at test
         time, to finalize predictions. The logic for the decoder part of the encoder-decoder lives
         within the `forward` method.
-
         This method trims the output predictions to the first end symbol, replaces indices with
         corresponding tokens, and adds a field called `predicted_tokens` to the `output_dict`.
         """
         predicted_indices = output_dict["predictions"]
         if not isinstance(predicted_indices, numpy.ndarray):
             predicted_indices = predicted_indices.detach().cpu().numpy()
         all_predicted_tokens = []
-        for indices in predicted_indices:
+
+        for top_k_predictions in predicted_indices:
             # Beam search gives us the top k results for each source sentence in the batch
             # but we just want the single best.
-            if len(indices.shape) > 1:
-                indices = indices[0]
-            indices = list(indices)
-            # Collect indices till the first end_symbol
-            if self._end_index in indices:
-                indices = indices[: indices.index(self._end_index)]
-            predicted_tokens = [
-                self.vocab.get_token_from_index(x, namespace=self._target_namespace)
-                for x in indices
-            ]
-            all_predicted_tokens.append(predicted_tokens)
+            # if len(indices.shape) > 1:
+            #     indices = indices[0]
+            batch_predicted_tokens = []
+            for indices in top_k_predictions:
```

Comment:
Why is the extra loop necessary now?

Reply:
The original code only return top1 result for beam_search. It's not convenient if we want eval top5 score (or we want choose result by some hand-craft algorithm). So I use the code segment in [copy-net](https://github.com/allenai/allennlp-models/blob/master/allennlp_models/generation/models/copynet_seq2seq.py#L866) to get all results.